config/config.go

```diff
@@ -128,6 +129,46 @@ func (cfg *AuctionTimeouts) validate(errs configErrors) configErrors {
 	return errs
 }
 
+func (data *ExternalCache) validate(errs configErrors) configErrors {
+	if data.Host == "" {
+		if data.Path != "" {
+			// Throw an error because we have a blank Host nad a non-blank Path
+			errs = append(errs, fmt.Errorf("Could not parse URL with empty Host: %s \n", data.Host+data.Path))
+		}
+		// Blank host and path are valid and will not throw an error
+		return errs
+	}
+
+	// Add "/" suffix to the path if missing
+	if data.Path != "" && !strings.HasPrefix(data.Path, "/") && !strings.HasSuffix(data.Host, "/") {
+		data.Path = "/" + data.Path
+	}
+
+	// Add "//" suffix hosts that don't come with a scheme
+	if strings.Index(data.Host, "//") == -1 {
+		data.Host = "//" + data.Host
+	}
+
+	u, err := url.Parse(data.Host + data.Path)
```

Comment:
This is trying to correct a configuration error. It might be easier to identity incorrect input instead. Using url.parse is a good idea, but consider using it to ensure the host doesn't specify a protocol/scheme.

Reply:
Ok. Let me go back to formula (Sam Raimi Spiderman reference) for this one. Maybe we can discuss it in tomorrow's stand-up.

Reply:
Could we use the IsHost() and IsPath() validation methods from the govalidator package ?

Reply:
> Could we use the IsHost() and IsPath() validation methods from the govalidator package ?

IsHost() fails because the url can contain a port, IsPath() matches Windows paths too. There are premade regexes in the validator package that can be used.

Instead of validating host and path separately, we could go the other way and validate the final composed result, something like this?
```go
u, err := url.Parse("http://" + host + path)
if err == nil && (host != u.Host || path != u.path) {
    err = fmt.Errorf("Invalid external cache config host='%s' path='%s'", host, path)
}
```

Reply:
Updated and will still use `url.Parse` thank you @laurb9